### PR TITLE
doc: Building the GTK app on macOS

### PR DESCRIPTION
**Pull request by Coeur**
Thanks @nevack and @GaryElshaw for figuring out the steps.

### CODE REVIEW

File: docs/Building-Transmission.md [hunk 1]
```diff
@@ -2,15 +2,44 @@
...
 
+### Building the native app with Xcode ###
+Transmission has an Xcode project file for building in Xcode. 
+- Open Transmission.xcodeproj
+- Run the Transmission scheme
+
+### Building the native app with make ###
+Build the app:
+```console
+cmake -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Comment (nevack) [inline docs/Building-Transmission.md:25]: IMO, it's 2022.
Better use `Ninja` generator
like this:
```bash
cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo
ninja -C build transmission-mac
open build/macosx/Transmission.app
```
Ninja is also parallel by default, so much improved build time.

**Comment by mikedld** [comment]
For those reading this document, why is it important to build GTK client on Mac in particular? Why not e.g. Qt client (which at least provides additional functionality compared to Mac client)?

I (think I) understand the problem you're solving, but it seems a bit counter-intuitive. Given that the whole document is very much outdated, would you like reworking it further, so that common building techniques that don't depend on the OS (meaning, CMake-based builds) are documented separately, with OS-specific sections only mentioning the dependencies required for build to succeed?

**Comment by Coeur** [comment]
I would leave the reworking to you or someone with a Linux at disposal.
But installing gtk using `brew` is quite mac-centric, while on Linux one would likely use `apt-get` probably, so not that much can be made common. And it's easier to read just the part under your OS than to have to search different installation steps in fragmented instructions.

**Comment by GaryElshaw** [comment]
I flick between macOS, Ubuntu, and Manjaro Gnome on a daily basis; i also build the Qt client on macOS for testing purposes. Personally, i prefer the macOS client above all of the others for multiple reasons, and personally don't think the Qt client offers more features. 

One of the things i've discovered is that almost all bittorrent clients don't offer the ability to save a .torrent file, the macOS client does.

**Comment by Coeur** [comment]
Would you feel like sharing the installation instructions for Qt on macOS? (including how to install Qt itself)

**Comment by GaryElshaw** [comment]
Sure thing!

**Comment by ckerr** [comment]
This isn't the boldest answer, but I kind of agree with all of the above :slightly_smiling_face: 

@mikedld is right that thet doc is out-of-date and the right fix is to rewrite the whole thing. Until then, I'm OK with an interim update like this one for building the GTK client on macOS (and would also be OK with adding the Qt client too as described by @GaryElshaw above).

> Personally, i prefer the macOS client above all of the others for multiple reasons, and personally don't think the Qt client offers more features.

The main advantage to `transmission-qt` is the ability to talk to a remote transmission-daemon session

**Review by ckerr — APPROVED** [review]

**Comment by ckerr** [comment]
re-requesting review from @mikedld for clarification: are incremental PRs OK or would you rather skip that and go straight to a rewrite?

**Comment by mikedld** [comment]
I can't force people to do what they don't want. Just hoping we'll eventually get to it...

**Pull request merged**